xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-io/src/main/java/org/xwiki/annotation/io/internal/DefaultIOService.java

```diff
@@ -130,8 +131,7 @@ public void addAnnotation(String target, Annotation annotation) throws IOService
             EntityReference annotationClassReference = this.configuration.getAnnotationClassReference();
             annotationClassReference =
                 annotationClassReference.removeParent(annotationClassReference.extractReference(EntityType.WIKI));
-            int id = document.createXObject(annotationClassReference, deprecatedContext);
-            BaseObject object = document.getXObject(this.configuration.getAnnotationClassReference(), id);
+            BaseObject object = document.newXObject(annotationClassReference, deprecatedContext);
```

Comment:
`deprecatedContext` is a weird variable name?

Reply:
That code is old enough to still have the illusions that the XWikiContext would be replaced in a short future :)

Reply:
yes, I think I can rename it to `context` safely.

Reply:
I tend to prefer the name xcontext on my side for the XWikiContext (a little more explicit and more consistent with script variable names), but there is no official rule.